lib/spdy-transport/connection.js

```diff
@@ -281,7 +281,7 @@ Connection.prototype._handleFrame = function _handleFrame(frame) {
 
   // Session window update
   if (frame.type === 'WINDOW_UPDATE' && frame.id === 0) {
-    if (state.version < 3.1 && state.autoSpdy31) {
```

Comment:
So in the current case, with `autoSpdy31: true, plain: true`, SPDY/3.1 clients work fine. But with a SPDY/3 client, the server sends a `id=0 WINDOW_UPDATE` because of https://github.com/indutny/spdy-transport/blob/master/lib/spdy-transport/connection.js#L246-L247. Because SPDY/3 doesn't implement session-level flow-control, a SPDY/3 client may treat that frame as a protocol error/violation and drop the connection (as iSPDY does because it fails to locate a stream with id=0 and decides to drop the connection: https://github.com/Voxer/iSPDY/blob/master/src/ispdy.m#L1227-L1242).

I've pushed a fix for this issue, by adding ` && state.version >= 3.1` to the checks. This ensures that the `id=0 WINDOW_UPDATE` is only sent for SPDY/3.1 clients when `autoSpdy31` is set to `true`.

Without this additional check, the server sends `id=0 WINDOW_UPDATE` to a SPDY/3 client causing it to disconnect.